services/knack_location_updater.py

```diff
@@ -0,0 +1,357 @@
+        data = []
+        if record[layer["updateFields"]] != data:
+            record[layer["updateFields"]] = data
+            changed = True
+
+    return record, changed
+
+def handle_features(changed, layer, record, res):
+    """
+    Handling the case where have features returned from AGOl
+
+    Parameters
+    ----------
+    changed: bool
+        Flag that tells us that we changed something in this record
+    layer : dict
+        The config information of the layer queried in AGOl.
+    record: dict
+        The knack record we are potentially editing.
+    res: dict
+        The response from the AGOL endpoint
+
+    Returns
+    -------
+    record: dict
+        Edited knack record
+    changed: bool
+        Flag that tells us that we changed something in this record
+    """
+
+    # Case for config where we only use one feature
+    if layer["handle_features"] == "use_first":
+        #  use first feature in results and join feature data to location record
+        feature = res["features"][0]
+        data = str(feature["attributes"][layer["outFields"]]).strip()
+
+        if record[layer["updateFields"]] != data:
+            logger.info(
+                f"Change Detected for ID:{record['id']}:{layer['outFields']} from: {record[layer['updateFields']]} to {data}"
+            )
+            record[layer["updateFields"]] = str(
+                feature["attributes"][layer["outFields"]]
+            )
+            changed = True
+    # Case where we use all returned features
+    elif layer["handle_features"] == "merge_all":
+        #  concatenate feature attributes from each feature and join to record
+        features = res["features"]
+
+        data = []
+        for feature in features:
+            data.append(
+                str(feature["attributes"][layer["outFields"]]).strip()
+            )
+
+        if layer["apply_format"]:
+            data = format_stringify_list(data)
+        else:
+            record[layer["updateFields"]] = record[
+                layer["updateFields"]
+            ].split(",")
+            record[layer["updateFields"]] = [
+                item.strip() for item in record[layer["updateFields"]]
+            ]
+
+        if record[layer["updateFields"]] != data:
+            logger.info(
+                f"Change Detected for ID:{record['id']}:{layer['outFields']} from: {record[layer['updateFields']]} to {data}"
+            )
+            record[layer["updateFields"]] = data
+            changed = True
+
+    return record, changed
+
+
+
+def local_timestamp():
+    """
+    Create a "local" timestamp (in milliseconds), ie local time represented as a unix timestamp.
+    Used to set datetimes when writing Knack records, because Knack assumes input
+    time values are in local time. Note that when extracing records from Knack,
+    timestamps are standard unix timestamps in millesconds (timezone=UTC).
+    """
+    return arrow.now().replace(tzinfo="UTC").timestamp * 1000
```

Comment:
@Charlie-Henry this assumes that the script is running on a machine that's configured on US/Central time. I think this small change ensure we always get the TZ shift right:

```python
return arrow.now("US/Central").replace(tzinfo="UTC").timestamp * 1000
```

Also, I was completely wrong when I wrote this bit of the docstring:
> Note that when extracing records from Knack,
    timestamps are standard unix timestamps in millesconds (timezone=UTC).

That's [false](https://cityofaustin.github.io/knackpy/docs/user-guide/#timestamps-and-localization) and you can just remove that sentence.